src/rpp/rpp/operators/retry_when.hpp

```diff
@@ -0,0 +1,194 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/observables/fwd.hpp>
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/strategy.hpp>
+#include <rpp/utils/constraints.hpp>
+#include <rpp/utils/utils.hpp>
+
+namespace rpp::operators::details
+{
+    template<rpp::constraint::observer     TObserver,
+             rpp::constraint::observable   TObservable,
+             rpp::constraint::decayed_type TNotifier>
+    struct retry_when_state : public rpp::composite_disposable
+    {
+        template<typename Observer, typename Observable, typename Notifier>
+        retry_when_state(Observer&& observer, Observable&& observable, Notifier&& notifier)
+            : observer(std::forward<Observer>(observer))
+            , observable(std::forward<Observable>(observable))
+            , notifier(std::forward<Notifier>(notifier))
+        {
+        }
+
+        RPP_NO_UNIQUE_ADDRESS TObserver   observer;
```

Comment:
first member with RPP_NO_UNIQUE_ADDRESS for class with base class with vtable could cause UB on MSVC. place bool as first member or remove it for first member pls